spec/unit/multi/advance_spec.rb

```diff
@@ -122,4 +122,53 @@
       restore
     ].join)
   end
+
+  it "advances progress bars correctly with indeterminate children" do
+    bars = described_class.new("total: :current", output: output)
+
+    bar1 = bars.register("one: :current")
+    bar2 = bars.register("two: :current")
+
+    bar1.advance
+
+    output.rewind
+    expect(output.read).to eq([
+      "\e[1G#{top}total: 1\n",
+      "\e[1G#{bottom}one: 1\n"
+    ].join)
+
+    bar2.advance
+
+    output.rewind
+    expect(output.read).to eq([
+      "\e[1G#{top}total: 1\n",
+      "\e[1G#{bottom}one: 1\n",
+      save,
+      "\e[2A", # up 2 lines
```

Comment:
I'm in favour of adding helpers to make the specs more readable. The way I would approach adding reusable pieces of functionality is by adding another file to `spec/support`. Inside a file you can extend RSpec by registering a module:

```ruby
module Support
  module XYZ # after the file name
    def move_up_and_write(...)
       ...
    end
  end
end

RSpec.configure do |config|
   config.include(Support::XYZ)
end
```
This will expose the defined functions automatically in all the specs.

However, my preference would be to address such improvements in a separate PR.